packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.tsx

```diff
@@ -0,0 +1,126 @@
+import { debounce } from 'lodash';
+import React, { useCallback, useMemo, useRef } from 'react';
+
+import { Field, FieldProps } from '../Forms/Field';
+import { InlineToast } from '../InlineToast/InlineToast';
+
+/**
+1.- Use the Input component as a base
+2.- Just save if there is any change and when it loses focus
+3.- Set the loading to true while the backend is saving
+4.- Be aware of the backend response. If there is an error show a proper message and return the focus to the input.
+5.- Add aria-live="polite" and check how it works in a screen-reader.
+Debounce instead of working with onBlur?
+import debouncePromise from 'debounce-promise';
+or
+import { debounce} from 'lodash';
+ */
+
+export interface Props extends FieldProps {
+  //Function to be run onBlur or when finishing writing
+  onFinishChange: (inputValue: string | number | readonly string[] | undefined) => Promise<void>;
```

Comment:
same here... this `inputValue` should be type `T` and be passed the generic from `GenericProps`.